src/app/qgisapp.cpp

```diff
@@ -5746,6 +5758,48 @@ QgsVectorTileLayer *QgisApp::addVectorTileLayerPrivate( const QString &url, cons
   return layer.release();
 }
 
+QgsPointCloudLayer *QgisApp::addPointCloudLayerPrivate( const QString &uri, const QString &baseName, bool guiWarning )
```

Comment:
QgisApp::openLayer will also need to call addPointCloudLayerPrivate to allow direct drag and drop of point cloud sources to the canvas from file managers.